scripts/local_cloudbuild_test.py

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test for local_cloudbuild.py"""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import unittest
+import unittest.mock
+
+import yaml
+
+import local_cloudbuild
+
+
+class ValidationUtilsTest(unittest.TestCase):
```

Comment:
Practicality: This code works out of the box on a corp workstation, and stock Ubuntu, which is a goal for me.  Adding a py.test dependency (or yaml schema validation library instead of my adhoc stuff above) would make this no longer true.

Style-wise: I'm completely biased here since I wrote googletest, but I'm not going to use py.test.  Too much implicit, too much magic, and other things I don't like.

At least you didn't ask me to use BDD.  Bleah.